assets/js/blocks/attribute-filter/block.js

```diff
@@ -188,17 +184,16 @@ const AttributeFilterBlock = ( {
 	}, [ productAttributesQuery, attributeObject.taxonomy ] );
 
 	const currentCheckedQuery = useShallowEqual( checkedQuery );
-
+	const previousCheckedQuery = usePrevious( currentCheckedQuery );
 	// Track ATTRIBUTES QUERY changes so the block reflects current filters.
-	useEffect(
-		() => {
-			if ( ! isShallowEqual( checked, checkedQuery ) ) {
-				setChecked( checkedQuery );
-			}
-		},
-		// We only want to apply this effect when the query changes, so we are intentionally leaving `checked` out of the dependencies.
-		[ currentCheckedQuery ]
-	);
+	useEffect( () => {
+		if (
+			! isShallowEqual( previousCheckedQuery, currentCheckedQuery ) && // checked query changed
```

Comment:
We only want to run this effect when `currentCheckedQuery` changes, but since we added new deps, we need to ensure the `currentCheckedQuery` is not equal to the previous one.